geonetwork/gn3-migration/schema-validation/fix-mcp-schema-validation.xsl

```diff
@@ -0,0 +1,318 @@
+        <gmd:date />
+      </xsl:if>
+    </xsl:copy>
+  </xsl:template>
+
+    <!-- 2a044b8f-249a-4ed4-bfb7-20f49d563811 -->
+  <!-- Removed unnecessary `<mcp:beginTime>` element   -->
+  <xsl:template match="gmd:temporalElement/mcp:EX_TemporalExtent/mcp:beginTime[./gco:DateTime[not(*) and not(normalize-space())]]" xmlns:mcp="http://bluenet3.antcrc.utas.edu.au/mcp"/>
+
+
+  <!--  /tmp/gn-validate/9e1414b2-3f8e-47c4-85df-dcc84444c325/metadata/metadata.xml:521: element Decimal: Schemas validity error : Element '{http://www.isotc211.org/2005/gco}Decimal': '-17.91.73' is not a valid value of the atomic type 'xs:decimal'.-->
+  <xsl:template match="gmd:geographicElement/gmd:EX_GeographicBoundingBox/gmd:southBoundLatitude/gco:Decimal[./text()='-17.91.73']">
+    <gco:Decimal>
+      <xsl:apply-templates select="@*"/>
+      <xsl:value-of select="number('-17.9173')" />
+    </gco:Decimal>
+  </xsl:template>
+  <!--  /tmp/gn-validate/9f8e77e3-8ea8-44cb-96b5-3cd2efb38e51/metadata/metadata.xml:591: element Decimal: Schemas validity error : Element '{http://www.isotc211.org/2005/gco}Decimal': '147.53.08' is not a valid value of the atomic type 'xs:decimal'.-->
+  <!--  /tmp/gn-validate/e1350e76-4e55-4fb6-856a-b9cb5108d88c/metadata/metadata.xml:591: element Decimal: Schemas validity error : Element '{http://www.isotc211.org/2005/gco}Decimal': '147.53.08' is not a valid value of the atomic type 'xs:decimal'.-->
+  <xsl:template match="gmd:geographicElement/gmd:EX_GeographicBoundingBox/gmd:eastBoundLongitude/gco:Decimal[./text()='147.53.08']">
+    <gco:Decimal>
+      <xsl:apply-templates select="@*"/>
+      <xsl:value-of select="number('147.5308')" />
+    </gco:Decimal>
+  </xsl:template>
+
+  <!--  <xsl:template match="gmd:spatialResolution/gmd:MD_Resolution/gmd:distance/gco:Distance[not(number(./text()))]">-->
+<!--    <gco:Distance>-->
+<!--      <xsl:apply-templates select="@*"/>-->
+<!--      <xsl:value-of select="number(substring(./text(), 1, 1))" />-->
+<!--    </gco:Distance>-->
+<!--  </xsl:template>-->
+
+  <!--  7 element Distance: Schemas validity error : Element '{http://www.isotc211.org/2005/gco}Distance': '1.5 Km range resolution determined by radar bandwidth' is not a valid value of the atomic type 'xs:double'.-->
+  <xsl:template match="gmd:spatialResolution/gmd:MD_Resolution/gmd:distance/gco:Distance[./text()='1.5 Km range resolution determined by radar bandwidth']">
+    <gco:Distance>
+      <xsl:apply-templates select="@*"/>
+      <xsl:value-of select="number('1.5')" />
+    </gco:Distance>
+  </xsl:template>
+
+  <!--  11 element Distance: Schemas validity error : Element '{http://www.isotc211.org/2005/gco}Distance': '3 Km range resolution determined by radar bandwidth' is not a valid value of the atomic type 'xs:double'.-->
+  <xsl:template match="gmd:spatialResolution/gmd:MD_Resolution/gmd:distance/gco:Distance[./text()='3 Km range resolution determined by radar bandwidth']">
+    <gco:Distance>
```

Comment:
I am still encountering issues for the `<gco:Distance>` it appears that the `<gco:Distance>` object needs to be wrapped in a `<gco:distance>` tag rather than replace with one.
Current:
```
      <gmd:spatialResolution>
        <gmd:MD_Resolution>
          <gmd:distance uom="kilometre">6</gmd:distance>
        </gmd:MD_Resolution>
      </gmd:spatialResolution>
```
Valid:
```
      <gmd:spatialResolution>
        <gmd:MD_Resolution>
          <gmd:distance>
            <gco:Distance uom="kilometre">6</gco:Distance>
          </gmd:distance>
        </gmd:MD_Resolution>
      </gmd:spatialResolution>
```